query/engine.go

```diff
@@ -99,6 +125,10 @@ func (b LocalQueryBuilder) Execute(ctx context.Context, callback func(r arrow.Re
 		logicalPlan = optimizer.Optimize(logicalPlan)
 	}
 
+	for _, optimizer := range b.optimizations {
```

Comment:
nit: add a comment about the difference between these optimizations and the ones above